plugin/http/httptrace/httptrace.go

```diff
@@ -59,10 +62,38 @@ func (t *Transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	resp, err := t.base().RoundTrip(req)
 
 	// TODO(jbd): Add status and attributes.
-	trace.EndSpan(ctx)
+	if err != nil {
+		trace.EndSpan(ctx)
+		return resp, err
+	}
+
+	// trace.EndSpan(ctx) will be invoked after
+	// resp.Body.Close() has been invoked.
+	resp.Body = &bodyProxyReader{rc: resp.Body, ctxToEnd: ctx}
 	return resp, err
 }
 
+type bodyProxyReader struct {
+	rc        io.ReadCloser
+	closeOnce sync.Once
+	ctxToEnd  context.Context
+}
+
+var _ io.ReadCloser = (*bodyProxyReader)(nil)
+
+func (bpr *bodyProxyReader) Read(b []byte) (int, error) {
+	return bpr.rc.Read(b)
+}
+
+func (bpr *bodyProxyReader) Close() error {
+	var err error
+	bpr.closeOnce.Do(func() {
```

Comment:
is this necessary? seems like if the caller calls close more than once they are at fault. Is this just a workaround for #328? If so I think it would be better to fix the root cause.

Reply:
@ramonza good questions. I wasn't thinking about #328 when I wrote that, it is just a precaution that I like to have to preserve `trace.EndSpan(ctx)` being invoked once as it was before.